libs/wingc/src/parser.rs

```diff
@@ -260,6 +261,42 @@ fn parse_wing_file(
 	dependent_wing_paths
 }
 
+fn dir_contains_wing_file(dir_path: &Utf8Path) -> bool {
+	for entry in fs::read_dir(dir_path).unwrap() {
+		let entry = entry.unwrap().path();
+		let path = Utf8Path::from_path(&entry).unwrap();
+
+		if path.is_dir() {
+			if dir_contains_wing_file(&path) {
+				return true;
+			}
+		} else if path.extension() == Some("w") {
+			return true;
+		}
+	}
+	false
+}
+
+fn contains_non_symbolic(str: &str) -> bool {
+	// uses the same regex pattern from grammar.js for valid identifiers
+	let re = Regex::new(r"^([A-Za-z_$][A-Za-z_$0-9]*|[A-Z][A-Z0-9_]*)$").unwrap();
```

Comment:
```suggestion
	let re = Regex::new(r"^([A-Za-z_][A-Za-z_0-9]*|[A-Z][A-Z0-9_]*)$").unwrap();
```

let's not allow `$` in directory names